lib/vrl/tests/src/docs.rs

```diff
@@ -0,0 +1,162 @@
+use serde::Deserialize;
+use std::collections::{BTreeMap, HashMap};
+use std::fs;
+use std::process::Command;
+// use vrl::Value;
+use serde_json::{Map, Value};
+
+use crate::Test;
+
+/// A list of function examples that should be skipped in the test run.
+///
+/// This mostly consists of functions that have a non-deterministic result.
+const SKIP_FUNCTION_EXAMPLES: &[&str] = &[
+    "uuid_v4",
+    "strip_ansi_escape_codes",
+    "get_hostname",
+    "now",
+    "get_env_var",
+];
+
+#[derive(Debug, Deserialize)]
+pub struct Reference {
+    examples: Vec<Example>,
+    functions: HashMap<String, Examples>,
+    expressions: HashMap<String, Examples>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Examples {
+    examples: Vec<Example>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Example {
+    title: String,
+    #[serde(default)]
+    input: Option<Event>,
+    source: String,
+    #[serde(rename = "return")]
+    returns: Option<Value>,
+    output: Option<Event>,
+    raises: Option<Error>,
+}
+
+#[derive(Debug, Default, Deserialize)]
+#[serde(default)]
+pub struct Event {
+    log: Map<String, Value>,
+
+    // TODO: unsupported for now
+    metric: Option<Map<String, Value>>,
+}
+
+#[derive(Debug, Default, Deserialize)]
+#[serde(default)]
+pub struct Error {
+    compiletime: String,
+}
+
+pub fn tests() -> Vec<Test> {
+    let dir = fs::canonicalize("../../../scripts").unwrap();
+
+    let output = Command::new("bash")
+        .current_dir(dir)
+        .args(&["cue.sh", "export", "-e", "remap"])
+        .output()
+        .expect("failed to execute process");
```

Comment:
A bit hacky, but it works 🤷

Reply:
If we aren't using anything explicitly `bash`, `sh` would probably be better.

Reply:
It requires bash to run.